lib/ContractInteract/BrandedToken.js

```diff
@@ -0,0 +1,416 @@
+   *             is set.
+   * @param decimals The value to which tokenDecimals, defined in EIP20Token,
+   *                 is set.
+   * @param conversionRate The value to which conversionRate is set.
+   * @param conversionRateDecimals The value to which
+   *                               conversionRateDecimals is set.
+   * @param organization Organization contract address.
+   *
+   * @returns {Object} Raw transaction.
+   */
+  static deployRawTx(
+    web3,
+    valueToken,
+    symbol,
+    name,
+    decimals,
+    conversionRate,
+    conversionRateDecimals,
+    organization,
+  ) {
+    if (!(web3 instanceof Web3)) {
+      throw new TypeError(
+        `Mandatory Parameter 'web3' is missing or invalid: ${web3}`,
+      );
+    }
+    if (!Web3.utils.isAddress(valueToken)) {
+      throw new TypeError(`Invalid valueToken address: ${valueToken}.`);
+    }
+    if (!Web3.utils.isAddress(organization)) {
+      throw new TypeError(`Invalid organization address: ${organization}.`);
+    }
+    if (!(conversionRate > 0)) {
+      throw new TypeError(`Invalid conversion rate: ${conversionRate}. It should be greater than zero`);
+    }
+    if (!(conversionRateDecimals < 5)) {
+      throw new TypeError(`Invalid conversion rate decimal: ${conversionRateDecimals}. It should be less than 5`);
+    }
+
+    const abiBinProvider = new AbiBinProvider();
+    const bin = abiBinProvider.getBIN(ContractName);
+
+    const args = [
+      valueToken,
+      symbol,
+      name,
+      decimals,
+      conversionRate,
+      conversionRateDecimals,
+      organization,
+    ];
+
+    const contract = Contracts.getBrandedToken(web3);
+
+    return contract.deploy(
+      {
+        data: bin,
+        arguments: args,
+      },
+    );
+  }
+
+  /**
+   * This calculates branded tokens equivalent to given value tokens.
+   *
+   * @param valueTokens Amount of value token.
+   *
+   * @return {Promise<string>} Promise that resolves to amount of branded token.
+   */
+  convertToBrandedTokens(valueTokens) {
+    return this.contract.methods
+      .convertToBrandedTokens(valueTokens)
+      .call();
+  }
+
+  /**
+   * Request stake for given amount. Approval for stake amount to branded
+   * token is required before calling this method.
+   *
+   * @param stakeAmount Stake amount.
+   * @param txOptions Transaction options.
+   *
+   * @return {Promise<Object>} Promise that resolves to transaction receipt.
+   */
+  async requestStake(stakeAmount, txOptions) {
+    if (!txOptions) {
+      const err = new TypeError(`Invalid transaction options: ${txOptions}.`);
+      return Promise.reject(err);
+    }
+    if (!Web3.utils.isAddress(txOptions.from)) {
+      const err = new TypeError(
+        `Invalid from address ${txOptions.from} in transaction options.`,
+      );
+      return Promise.reject(err);
+    }
+
+    const mintedAmount = await this.convertToBrandedTokens(stakeAmount);
```

Comment:
I believe requestStake interface should be aligned with BrandedToken.requestStake.
So mintAmount should not be calculated in this function. It causes an extra geth call as well changes the interfaces. Staker can call convertToBrandedTokens to know the mint amount.